src/appleseed/renderer/kernel/lighting/sppm/sppmlightingengine.cpp

```diff
@@ -291,6 +292,28 @@ namespace
                 // Don't compute lighting in the first pass if importons are enabled.
                 if (!m_params.m_enable_importons || m_pass_callback.get_pass_number() > 0)
                 {
+                    for (auto& light_info : m_backward_light_sampler.get_non_physical_light_vector())
+                    {
+                        Spectrum value;
+                        light_info.m_light->evaluate(
+                            m_shading_context,
+                            normalize(-Vector3d(vertex.m_outgoing.get_value())),
+                            value);
+
+                        // Apply path throughput.
+                        value *= vertex.m_throughput;
+
+                        // Optionally clamp secondary rays contribution.
+                        if (m_params.m_path_tracing_has_max_ray_intensity && vertex.m_path_length > 1 && vertex.m_prev_mode != ScatteringMode::Specular)
```

Comment:
We shouldn't clamp the intensity of each individual light, rather we should clamp the sum of all lights + the environment.